### PR TITLE
Handle malformed license signatures

### LINKED ISSUE #35340: buffer_underflow_exception when putting invalid license
**Elasticsearch version**: 6.4.2

**Plugins installed**: []

**JVM version** (`java -version`): 1.8.0_191

**OS version** (`uname -a` if on a Unix-like system): Linux 4.15.0-38-generic #41-Ubuntu SMP Wed Oct 10 10:59:38 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

**Description of the problem including expected versus actual behavior**:

When adding an invalid license, a buffer underflow error results.  I presume that the cause is some part of the license being invalid, but that the error could/should be better caught and the response message could be clearer.
The error resulted from using the example [here](https://www.elastic.co/guide/en/elasticsearch/reference/current/update-license.html) without putting in the actual license.

**Steps to reproduce**:

1. Put the license:
```
PUT _xpack/license
{
  "licenses": [
    {
      "uid":"893361dc-9749-4997-93cb-802e3d7fa4xx",
      "type":"basic",
      "issue_date_in_millis":1411948800000,
      "expiry_date_in_millis":1914278399999,
      "max_nodes":1,
      "issued_to":"issuedTo",
      "issuer":"issuer",
      "signature":"xx"
    }
    ]
}
```

2. Results in response:

```
{
  "error": {
    "root_cause": [
      {
        "type": "buffer_underflow_exception",
        "reason": null
      }
    ],
    "type": "buffer_underflow_exception",
    "reason": null
  },
  "status": 500
}
```

### PR DESCRIPTION
**Pull request by jkakavas**
This commit adds a more user friendly error message when a license
signature is malformed/truncated in a way that it cannot be
meaningfully parsed

Resolves #35340

### CODE REVIEW

File: x-pack/plugin/core/src/main/java/org/elasticsearch/license/License.java [hunk 1]
```diff
@@ -469,9 +470,17 @@ public static License fromXContent(XContentParser parser) throws IOException {
...
-            int version = byteBuffer.getInt();
+            int version;
+            // In case the signature is truncated/malformed we might end up with fewer than 4 bytes in the byteBuffer
+            // or with a string that cannot be base64 decoded. In either case return a more friendly error instead of
+            // just throwing the BufferUnderflowException or the IllegalArgumentException
+            try {
+                byte[] signatureBytes = Base64.getDecoder().decode(builder.signature);
+                ByteBuffer byteBuffer = ByteBuffer.wrap(signatureBytes);
+                version = byteBuffer.getInt();
+            } catch (BufferUnderflowException | IllegalArgumentException e) {
```
Comment (tvernum) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/license/License.java]: Is it intentional that you don't just catch `Exception`? I don't think there's any exception that could be thrown here that we wouldn't want to wrap with a better message.

Comment (jkakavas) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/license/License.java]: Yes, I did it on purpose. `decode()` throws an `IllegalArgumentException` and `getInt()` throws a `BufferUnderflowException()` . I think that `wrap()` can't throw an `IndexOutOfBoundsException` when called with a byte array param. Am I missing any ?

Comment (jaymode) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/license/License.java]: I think it is safe to catch `Exception` here. Any exception that we get here would pretty much indicate a malformed signature right?

Comment (jkakavas) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/license/License.java]: Yeap, I'll adjust this

**Review by jaymode — APPROVED** [review]
LGTM

**Review by tvernum — APPROVED** [review]

**Pull request merged**